### PR TITLE
feat: add IST adapter

### PR DESCRIPTION
**Pull request by mujahidkay**
adds IST adapter for Agoric's inter-protocol

**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama):
Inter Protocol

##### Twitter Link:
https://x.com/inter_protocol

##### Website Link:
https://inter.trade

##### Logo (High resolution, will be shown with rounded borders):
Inter-Protocol-logo-symbol-color

##### Current TVL:
5.46 M

##### Chain:
Agoric

##### Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)
inter-stable-token

##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)
inter-stable-token

##### Short Description (to be shown on DefiLlama):
Inter Protocol is a decentralized stable token protocol on Agoric, issuing IST, an over-collateralized, multi-collateralized stable token designed for secure, cross-chain transactions in the Cosmos ecosystem.

##### Token address and ticker if any:
Ticker is $IST

##### Category (full list at https://defillama.com/categories) *Please choose only one: 
CDP

##### Oracle Provider(s): Specify the oracle(s) used (e.g., Chainlink, Band, API3, TWAP, etc.):
Independent Chainlink DON

##### Implementation Details: Briefly describe how the oracle is integrated into your project:
The price feeds powering Inter Protocol are operated by an independently run and managed DON, including 5 node operators that are also validators on the Agoric network. These are P2P, Simply Staking, 01node, Stakin and DSRV.
Price updates are supplied via open source Chainlink software to a custom smart contract implementation of Chainlink’s aggregator contracts compatible with Agoric’s javascript platform.

##### Documentation/Proof: Provide links to documentation or any other resources that verify the oracle's usage:
https://inter.trade/oracle-network/
https://agoric.com/blog/announcements/agoric-chainlink-integration/
https://github.com/SimplyStaking/agoric-cl-setup-docs
https://github.com/SimplyStaking/agoric-cl-middleware
https://community.agoric.com/t/inter-protocol-vaults-contract-implementations/261/3 

##### methodology (what is being counted as tvl, how is tvl being calculated):
Queries vault manager metrics, board auxes and oracle prices from agoric's subquery API. 
Fetches total collateral from vault manager metrics, decimal places from board auxes and computes typeOut/In ratio from oracle prices. 
In short, sums the following for each collateral:
```
(typeOutAmount * collateralValue) / (typeInAmount * (10^decimalPlaces))
```

### CODE REVIEW

**Comment by llamabutler** [comment]
The adapter at projects/agoric exports TVL: 
        
 
 
    agoric                    5.47 M
    
    total                    5.47 M

**Comment by llamabutler** [comment]
The adapter at projects/agoric exports TVL: 
        
 
 
    agoric                    5.07 M
    
    total                    5.07 M
